react/components/ProductAssemblyContext/Group.tsx

```diff
@@ -69,14 +78,42 @@ const initState = (assemblyOption: AssemblyOptionGroupState) => {
   return assemblyOption
 }
 
+const createRecursiveDispatch = ({
+  hasParent,
+  dispatch,
+}: {
+  hasParent: boolean
+  dispatch: Dispatch<DispatchAction>
+}) => {
+  if (hasParent) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const parentDispatch = useContext(ProductAssemblyDispatchContext)
```

Comment:
The rules of hooks should not be ignored in this case.

Hooks can't be created inside an `if`.

Reply:
Fixed